core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -2428,13 +2372,21 @@ private static void linkFile(FilesFacade ff, LPSZ from, LPSZ to) {
         if (ff.exists(from)) {
             if (ff.hardLink(from, to) == FILES_RENAME_OK) {
                 LOG.debug().$("renamed [from=").$(from).$(", to=").$(to).I$();
-            } else {
-                throw CairoException.critical(ff.errno())
-                        .put("could not create hard link [errno=").put(ff.errno())
-                        .put(", from=").put(from)
-                        .put(", to=").put(to)
-                        .put(']');
+                return;
+            } else if (ff.exists(to)) {
+                LOG.info().$("rename destination file exists, assuming failed rename attempt [path=").$(to).I$();
```

Comment:
nit: this might be a bit less confusing
```suggestion
                LOG.info().$("rename destination file exists, assuming previously failed rename attempt [path=").$(to).I$();
```